src/signatureverifier.cpp

```diff
@@ -0,0 +1,370 @@
+/*
+ *  This file is part of WinSparkle (https://winsparkle.org)
+ *
+ *  Copyright (C) 2009-2017 Vaclav Slavik
+ *  Copyright (C) 2017 Ihor Dutchak
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a
+ *  copy of this software and associated documentation files (the "Software"),
+ *  to deal in the Software without restriction, including without limitation
+ *  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ *  and/or sell copies of the Software, and to permit persons to whom the
+ *  Software is furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ *  DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+#include "signatureverifier.h"
+
+#include <stdexcept>
+
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#include <Wincrypt.h>
+
+#pragma comment(lib, "crypt32.lib")
```

Comment:
Missed this - it is for Base64ToBin.
Does WinSparkle support MinGW/Cygwin builds?

Reply:
Somewhat — there's the cmake build system that is meant to work with it, but it's not maintained by me. In any case, that's a trivial thing to `#ifdef`, so don't worry about that.

Reply:
I just don't feel right, if I keep it as is. [9d811dd](https://github.com/Youw/winsparkle/commit/9d811dd2840ce6795d17ca870514a01d523834a2) =)

Reply:
BTW: cmake build is broken after one of the expat updates

Reply:
@Youw Pro-tip: when you find yourself writing "BTW" in an issue tracker, stop right there — you're being off-topic. If you have a *completely unrelated* issue, file an unrelated issue in the tracker. In the case of cmake build, make that a PR (see the [explanation](https://github.com/vslavik/winsparkle/tree/master/cmake) why).

Reply:
@Youw I'm sorry, let me rephrase: since the cmake build is unsupported, you don't have to worry about it as part of this PR.

Reply:
Sure, totally agree.